pmoves/services/flute-gateway/main.py

```
@@ -1003,6 +1051,201 @@ async def get_persona(persona_id: str) -> Dict[str, Any]:
+    try:
+        result = await cloning_provider.get_training_status(persona_id)
+        REQUESTS_TOTAL.labels(endpoint="/v1/voice/clone/status", status="200").inc()
+        return VoiceCloneStatusResponse(**result)
+
+    except ValueError as exc:
+        REQUESTS_TOTAL.labels(endpoint="/v1/voice/clone/status", status="404").inc()
+        raise HTTPException(status_code=404, detail=str(exc)) from exc
+    except Exception:
+        REQUESTS_TOTAL.labels(endpoint="/v1/voice/clone/status", status="500").inc()
+        logger.exception("Failed to get training status")
+        raise HTTPException(status_code=500, detail="Failed to get training status")
```

Ruff (0.14.10)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)